### DISCUSSION TITLE
Prisma & Supabase in Windows 11:  Hang at same point always...

### DISCUSSION THREAD

**Discussion by DavidDurlach** [prisma/prisma#25713] [Q&A] [answered]
### Question

Hello.

**Background:** I am a complete newbie — not to coding generally, but in terms of full stack web development, database use (in any context), etc.

**Issue:**  I have been trying to get a sample T3 project (Node.js + TypeScript etc. https://create.t3.gg/en/introduction) working.  As part of that, I have been communicating with Supabase database via both prisma and psql within Windows CMD and/or Powershell window.
Using psql, I have no problem connecting to database, creating table, etc.  However, via prisma, all commands — "npx prisma migrate dev --name ini", "npx prisma migrate status", "npx prisma db push", etc., all hang in the same general place — at the last line below:
…...
_prisma:getSchema Reading schema from single file D:\Test\prisma\schema.prisma +1ms
prisma:getConfig Using getConfig Wasm +1ms
prisma:getConfig config data retrieved without errors in getConfig Wasm +0ms_
(Hang…)

And no table created.

**My efforts so far:**  I have worked with Google, and AIs (Claude, Open AI 01, Perplexity), for over a full day, trying many different connection strings within .env file etc., with different switches at the end, all to no effect.  Hang occurs identically in same place regardless.

**Summary:**  It is some interaction between Prisma and Supabase because, as noted,  Psql & Supabase work fine.  (But apparently, from reading, just using that latter combo not a good long-term strategy for various reasons.)

Any ideas of cause and what to change/try/examine?

Thanks,
David

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @DavidDurlach 

Can you please share your schema file? Did you include both `DATABASE_URL` and `DIRECT_URL` in the `datasource` section of your schema? This is described in this [documentation](https://supabase.com/partners/integrations/prisma)
```prisma
datasource db {
  provider          = "postgresql"
  url               = env("DATABASE_URL")
  directUrl         = env("DIRECT_URL")
}
```
where
```
DATABASE_URL="" # Set this to the Transaction connection pooler string 
DIRECT_URL=""  # Set this to the Session connection pooler string 
```

**Reply by DavidDurlach** [reply]
Hi @ludralph

Thank you, thank you, thank you — that fixed it!

[I find it surprising that none of AIs, or template code (at least that I saw/noticed in my default Windows install — maybe I somehow missed...) mentioned that structure.  In that regard, thanks both for specific snip, and doc link from which it came.]

MOST appreciatively,
David

**Reply by ludralph** [reply]
You're welcome! 🙏
